packages/core/core/src/services/document-service/repository.ts

```diff
@@ -250,15 +251,18 @@ export const createContentTypeRepository: RepositoryFactoryMethod = (uid) => {
       }),
     ]);
 
-    // Delete all published versions
-    await async.map(publishedToDelete, (entry: any) => entries.delete(entry.id));
-
     // Transform draft entry data and create published versions
-    const publishedEntries = await async.map(draftsToPublish, (draft: unknown) =>
+    const publishedEntries = await async.map(draftsToPublish, (draft: any) =>
       entries.publish(draft, queryParams)
     );
 
+    await syncUnidirectionalRelations(uid, oldPublishedVersions, publishedEntries);
+
+    // Delete old published versions
+    await async.map(oldPublishedVersions, (entry: any) => entries.delete(entry.id));
+
     publishedEntries.forEach(emitEvent('entry.publish'));
+
     return { documentId, entries: publishedEntries };
   }
 
```

Comment:
More of a general thought remark, when deleting data we do not clear the unidirectional relations pointing to nothing anymore. Sth we need to think about as a "cleanup task done in the bg"

Reply:
can we rely on the cascade delete we currently have?

Reply:
in some cases yes but not for all db or polymorphics)